src/python/pants/option/config.py

```diff
@@ -187,7 +199,7 @@ def format_str(value: str) -> str:
             # that .format() does not try to improperly interpolate.
             escaped_str = value.replace("{", "{{").replace("}", "}}")
             new_style_format_str = re.sub(
-                pattern=r"%\((?P<interpolated>[a-zA-Z_0-9]*)\)s",
+                pattern=r"%\((?P<interpolated>[a-zA-Z_0-9.]+)\)s",
```

Comment:
The `*` seemed like unusable or at least undesirable flexibility - an empty string as the placeholder name?! I may be missing something here.

Reply:
I think you're right, it probably should always have been `+`.

Reply:
That sounds right to me. I think that I wrote this regex over two years ago and probably made that oversight